### ISSUE TITLE
Forward-port / Back-port TODO

### ISSUE THREAD

**Issue by rvalyi** [OCA/l10n-brazil#1578]
Forward port from 12.0 to 13.0/14.0

- [x] https://github.com/OCA/l10n-brazil/pull/1624
- [x] https://github.com/OCA/l10n-brazil/pull/1659
- [x] https://github.com/OCA/l10n-brazil/pull/1690 no modulo l10n_br_nfe #1674
- [x] https://github.com/OCA/l10n-brazil/pull/1759
- [x] https://github.com/OCA/l10n-brazil/pull/1780 no modulo l10n_br_nfe #1674
- [x] https://github.com/OCA/l10n-brazil/pull/1814
- [x] https://github.com/OCA/l10n-brazil/pull/1809
- [x] https://github.com/OCA/l10n-brazil/pull/1823
- [x] https://github.com/OCA/l10n-brazil/pull/1764
- [x] https://github.com/OCA/l10n-brazil/pull/1610
- [x] https://github.com/OCA/l10n-brazil/pull/1804 no modulo l10n_br_fiscal
- [x] https://github.com/OCA/l10n-brazil/pull/1804 no modulo l10n_br_account #1688 
- [x] https://github.com/OCA/l10n-brazil/pull/1815 no modulo l10n_br_fiscal
- [x] https://github.com/OCA/l10n-brazil/pull/1815 no modulo l10n_br_nfe
- [x] https://github.com/OCA/l10n-brazil/pull/1849 no modulo l10n_br_fiscal
- [x] https://github.com/OCA/l10n-brazil/pull/1849 no modulo l10n_br_account
- [x] https://github.com/OCA/l10n-brazil/pull/1853 no modulo l10n_br_fiscal
- [x] https://github.com/OCA/l10n-brazil/pull/1853 no modulo l10n_br_nfe
- [x] https://github.com/OCA/l10n-brazil/pull/1996
- [x] https://github.com/OCA/l10n-brazil/pull/1997
- [x] https://github.com/OCA/l10n-brazil/pull/2017
- [x] https://github.com/OCA/l10n-brazil/pull/2012  #2848 

Backport port from 14.0 to 12.0

- [x] https://github.com/OCA/l10n-brazil/pull/1595
- [x] https://github.com/OCA/l10n-brazil/pull/1597
- [x] https://github.com/OCA/l10n-brazil/pull/1668
- [x] https://github.com/OCA/l10n-brazil/pull/1680
- [x] https://github.com/OCA/l10n-brazil/pull/1700
- [x] https://github.com/OCA/l10n-brazil/pull/1702
- [x] https://github.com/OCA/l10n-brazil/pull/1760
- [x] backport ? https://github.com/OCA/l10n-brazil/pull/2136

**Comment by rvalyi** [comment]
The idea behind this issue is to update it the same way as we do with Shopinvader https://github.com/shopinvader/odoo-shopinvader/issues/416 or how Acsone does it internally for mis-builder https://github.com/OCA/mis-builder/pulls?q=is%3Apr+is%3Aclosed+PORT .

Rationale: basically at Akretion we participate to the maintainance of around ~70% of the OCA/l10-brazil and some of our customers will still be on v12 in 2022. Yup. We won't take any v13 project, but we will soon focus on v14. Some core modules are easier to migrate with a v13 step than straight to v14, but as for us we will provide v13 PRs only for the very core modules (like l10n_br_account and l10n_br_nfe) and we won't maintain v13 actively. But of courset if others want to do it that's fine and they will easily be able to backport modules from v14 (rather than front-port from v12). As for ourselves, we think we will phase out 12.0 evolution at the same pace we will adopt v15 eventually (not before early 2022 at least).

It means we need a strict way to port the project PRs back and forth to ensure last features and bugfix are available accross versions and also ensure a smooth migration when upgrading instead of the module being totally different and possibly broken in version N+1. This is also why we should keep modules as simple as possible and prefer breaking things down in several small modules where people are then hold responsible for shit not to happen with the modules they maintain before anything else.

Also this is made possible because we ate the bullet on v12 by converting almost the whole codebase to the OCA pre-commit standards https://github.com/OCA/l10n-brazil/issues/1445. An ingrate work from the previous months but that is now paying when porting modules much more easily with relitively small diffs.

cc @marcelsavegnago @renatonlima @mbcosta @britoederr @netosjb @felipemotter @rpsjr @bmessiaz @luismalta @gabrielcardoso21 @ygcarvalh @marcos-mendez 

What do you guys think? I just wanted the main ideas were clear in plain English but answering in Portuguese is fine.

**Comment by marcos-mendez** [comment]
Pois é. Eu ainda to usando algumas coisas da 12 para a 14 então imagina pra 16. Continuo concordando como em 2021
